db/upgrade.php

```diff
@@ -413,5 +413,50 @@ function xmldb_cms_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2024090303, 'cms');
     }
 
+    if ($oldversion < 2024090304) {
+        // Update files belonging to mod_cms overview section content types to use the course module context id.
+        // Update the pathnamehash as well, otherwise files will display as missing until the content is edited and saved.
+        // Records are effectively copied, the old records remain in case anything relies on them and they can be used to 
+        // cross reference the new records if needed.
+
+        $sql = "SELECT f.*, ctx.id as ctxid
+                  FROM {files} f
+                  JOIN {customfield_data} cfd ON cfd.id = f.itemid
+                  JOIN {customfield_field} cff ON cff.id = cfd.fieldid AND cff.shortname = 'overview'
```

Comment:
I think the issue is for customfield which is using textarea, not only "overview".

Reply:
Correct, I have changed the sql to select cms types that use "fields" datasources.